app/src/App/index.js

```diff
@@ -49,17 +61,51 @@ const Home = () => {
         <h1>{process.env.REACT_APP_TITLE}</h1>
         <p>{process.env.REACT_APP_SUBTITLE}</p>
       </header>
-      {isAuthenticated ? <Flashcards /> : null}
+      {isAuthenticated && !loading ? (
+        <Flashcards flashcards={flashcards} />
+      ) : null}
     </>
   );
 };
 
-const Practice = () => {
-  const { isAuthenticated } = useAuth0();
+const Practice = ({ flashcards }) => {
+  console.log("flashcards", flashcards);
+  const [cardNumber, setCardNumber] = useState(0);
+  const [showFront, setShowFront] = useState(true);
+  const toggleSide = useCallback(() => {
+    setShowFront(!showFront);
+  }, [showFront]);
+  const incrementCard = useCallback(() => {
+    if (cardNumber === flashcards.length - 1) {
+      setCardNumber(0);
+    } else {
+      setCardNumber(cardNumber + 1);
+    }
+  }, [cardNumber]);
+  const decrementCard = useCallback(() => {
+    if (cardNumber === 0) {
+      setCardNumber(flashcards.length - 1);
+    } else {
+      setCardNumber(cardNumber - 1);
+    }
+  }, [cardNumber]);
 
   return (
     <>
-      <p>Front of the card</p>
+      {showFront && (
+        <Side
+          text={flashcards[cardNumber].front_of_card}
+          toggleSide={toggleSide}
+        />
+      )}
+      {!showFront && (
+        <Side
+          text={flashcards[cardNumber].back_of_card}
+          toggleSide={toggleSide}
+        />
+      )}
+      <button onClick={incrementCard}>Previous</button>
```

Comment:
What is the point of using useCallback if the function is called when the button is clicked?